src/Tasks/SGen.cs

```diff
@@ -253,9 +253,10 @@ protected override string GenerateCommandLineCommands()
                 // Append the references, if any.
                 if (References != null)
                 {
-                    foreach (string reference in References)
+                    if (References.Length > 0)
                     {
-                        commandLineBuilder.AppendSwitchIfNotNull("/reference:", reference);
+                        //Appending references if 
+                        commandLineBuilder.AppendSwitchIfNotNull("/r:", string.Join(",", References));
```

Comment:
I would prefer to avoid shortening `/reference:`; it's generally good style to use long-form command line arguments for readability and maintainability, and the real win here is not having to repeat it per `Reference`.

Reply:
I can leave full "/reference". Should I change this with the new commit within this request or create a new branch with one commit and recreate request?